### PR TITLE
build: Fix that custom docker image build

### PR DESCRIPTION
**Pull request by chadr123**
Commit Message:
The do_ci.sh builds the envoy and copies binary to under build-* directories.
But recently, the arm64 build system was added but it does not consider the
previous custom docker image build.
It is very useful to build the custom docker image and test it within k8s environment.

So, this patch support previous behavior again so that we can build the custom docker image.
If some more cpu architectures are added, we can simple extends more cpu architectures.

Signed-off-by: DongRyeol Cha <dr83.cha@samsung.com>
Additional Description: None
Risk Level: Low
Testing: bazel test and bazel build
Docs Changes: None
Release Notes: None

### CODE REVIEW

File: ci/build_setup.sh [hunk 1]
```diff
@@ -9,8 +9,10 @@ export PPROF_PATH=/thirdparty_build/bin/pprof
 [ -z "${NUM_CPUS}" ] && NUM_CPUS=`grep -c ^processor /proc/cpuinfo`
 [ -z "${ENVOY_SRCDIR}" ] && export ENVOY_SRCDIR=/source
 [ -z "${ENVOY_BUILD_TARGET}" ] && export ENVOY_BUILD_TARGET=//source/exe:envoy-static
+[ -z "${ENVOY_BUILD_ARCH}" ] && export ENVOY_BUILD_ARCH=`uname -m`
```
Comment (lizan) [inline ci/build_setup.sh:12]: style: use `"$(uname -m)"`

File: ci/do_ci.sh [hunk 2]
```diff
@@ -16,7 +16,17 @@ SRCDIR="${PWD}"
 . "$(dirname "$0")"/build_setup.sh $build_setup_args
 cd "${SRCDIR}"
 
+if [[ "${ENVOY_BUILD_ARCH}" == "x86_64" ]]; then
+  BUILD_ARCH_DIR="/linux/amd64"
+elif [[ "${ENVOY_BUILD_ARCH}" == "aarch64" ]]; then
+  BUILD_ARCH_DIR="/linux/arm64"
+else
+  echo "Unknown architecture for building: '${ENVOY_BUILD_ARCH}'"
+  exit 1;
```
Comment (lizan) [inline ci/do_ci.sh:25]: There are folks building on ppc64le / s390x, Shouldn't fail hard here. Just fall back to `linux/${ENVOY_BUILD_ARCH}`?

File: ci/build_setup.sh (hunk 1 above)
Comment (chadr123) [reply ci/build_setup.sh:12]: Ok. :)

File: ci/do_ci.sh (hunk 2 above)
Comment (chadr123) [reply ci/do_ci.sh:25]: NP. I will revise my patch.

**Comment by chadr123** [comment]
I have changed my patch. Please check it. :)

**Pull request merged**